drivers/counter/counter_ll_stm32_rtc.c

```diff
@@ -739,9 +725,6 @@ static DEVICE_API(counter, rtc_stm32_driver_api) = {
 	.start = rtc_stm32_start,
 	.stop = rtc_stm32_stop,
 	.get_value = rtc_stm32_get_value,
-#ifdef CONFIG_COUNTER_RTC_STM32_SUBSECONDS
```

Comment:
A few line above, i think `rtc_stm32_config::counter_info::max_top_value should be `UINT64_MAX` when `CONFIG_COUNTER_RTC_STM32_SUBSECONDS| is enabled:
```patch
 static const struct rtc_stm32_config rtc_config = {
 	.counter_info = {
+#ifdef CONFIG_COUNTER_RTC_STM32_SUBSECONDS
+		.max_top_value = UINT64_MAX,
+#else
 		.max_top_value = UINT32_MAX,
+#endif /* CONFIG_COUNTER_RTC_STM32_SUBSECONDS */